validate_states.py

```diff
@@ -433,11 +449,16 @@ def main(opt):
         outfile = os.path.join(outdir, filename)
         logger.info('Writing plot file %s' % outfile)
         plt.tight_layout()
+        plt.margins(.05)
         fig.savefig(outfile)
         plot['lines'] = filename
 
         if msid not in diff_only:
-            diff = tlm[msid][~bad_time_mask] - pred[msid][~bad_time_mask]
```

Comment:
Try to think about what is changing here due to the patch.  Is it the way `diff` is computed or is it the mask that is filtering the data?

Reply:
I don't understand the comment.  Do you see a bug or would you prefer I mask the diff directly (in an extra line) instead of masking the data as it is computed?

Reply:
Or would you like me to filter the data for these msids earlier (though that's awkward as I'd like it to plot) or is there something not clear about the comment?

Reply:
I'm trying to encourage more clear coding in general and not just copy/paste pieces of code.
```
ok = ~bad_time_mask  # Ignore bad times

# For these two validations, also ignore intervals during a dark current calibration
if msid in ['dither', 'pcad_mode']:
    ok &= ~dark_mask

diff = tlm[msid][ok] - pred[msid][ok]
```
The way you wrote this the two version of `diff` are not connected in any obvious way, except they turn out to be copy/paste from the original.

Reply:
Ah.  OK.  We might need a codeword for == "Try to think about how to improve style and readability in this area of the code", because I didn't pick up on it.  Thanks!  

I was too focused on guaranteeing that the code didn't change at all for the other "msids" (so the functional diffs really minimized that in the first try).

Reply:
Correctness will follow naturally from style and readability.  :smile: